### PR TITLE
serialize ipc message before write

### PR DESCRIPTION
**Pull request by artfwo**
this makes #22 non-reproducible for me probably due to single write operation: partial/torn messages don't end up in device_read_cb.

### CODE REVIEW

File: src/common/ipc.c [hunk 1]
```diff
@@ -72,65 +72,25 @@ read_strdata(int fd, size_t n, ...)
...
-#undef EMIT_VAR
-#undef EMIT_BYTES
-
-	return 0;
-}
-
 int
 sosc_ipc_msg_write(int fd, sosc_ipc_msg_t *msg)
 {
+	uint8_t buf[64];
```
Comment (wrl) [inline src/common/ipc.c]: i'd be more comfortable with bumping this up to 256

**Comment by wrl** [comment]
oh, that totally makes sense. yeah, just address the buffer length and i'll merge this

**Comment by artfwo** [comment]
@wrl what's wrong with the buffer length? too short?

File: src/common/ipc.c (hunk 1 above)
Comment (artfwo) [reply src/common/ipc.c]: 64 is used throughout the code for similar cases, do you want to increase it just here or everywhere? how about a named constant in that case?

Comment (wrl) [reply src/common/ipc.c]: good point actually. nevermind!

**Pull request merged**